src/cargo/ops/cargo_add/mod.rs

```diff
@@ -196,6 +197,13 @@ pub fn add(workspace: &Workspace<'_>, options: &AddOptions<'_>) -> CargoResult<(
         print_dep_table_msg(&mut options.config.shell(), &dep)?;
 
         manifest.insert_into_table(&dep_table, &dep)?;
+        if let Some(option) = dep.optional {
+            let rust_version_check =
+                check_rust_version_for_optional_dependency(options.spec.rust_version())?;
+            if option && rust_version_check {
```

Comment:
Can't we simplify the first if to `if dep.optional == Some(true)`?

Reply:
Thanks for the advice and solved.